### DISCUSSION TITLE
setValue() / addValue() type into adress bar instead of selected element

### DISCUSSION THREAD

**Discussion by batallo** [webdriverio/webdriverio#7938] [Q&A]
I'm using WebdriverIO + devtools:puppeteer + cucumber + Firefox Nightly.

When using **setValue()** / **addValue()**, the first letter of my input is typed into address bar, instead of selected element. The issue for same tests doesn't appear for mse or chrome browsers.

**Issue:**

[browserscreenshot][1]

After this, nothing happens until function timeouts

```
INFO devtools: COMMAND navigateTo("https://google.com/")
INFO devtools: RESULT null
INFO devtools: COMMAND findElement("css selector", "input[type=text]")
INFO devtools: RESULT { 'element-6066-11e4-a52e-4f735466cecf': 'ELEMENT-1' }
INFO devtools: COMMAND elementClear("ELEMENT-1")
INFO devtools: RESULT null
INFO devtools: COMMAND elementSendKeys("ELEMENT-1", "hello world")
```

**Code examples:**
Test:
```
Scenario: Try google
        When I open "google.com"
        Then I type "hello world" into "input[type=text]"
```
Steps:
```
When('I open {string}', async function (URL) {
    await browser.url(`https://${URL}`);
});

Then('I type {string} into {string}', async function (input, selector) {
    await $(selector).setValue(input);
});
```
Although there is a walkaround for some URLS with clicking on the element before using setValue(), this doesn't work for some cases (e.g. when redirecting from pre-login page to login page with pretyped-in login, I could not click + setValue for password field).

Hope anyone knows how this could be solved or walked around for all cases. Thanks.

  [1]: https://i.stack.imgur.com/NXhxv.png

**Comment by portyu9** [comment]
Hi @batallo 

I would remove the `devtools:puppeteer` + Firefox Nightly combination rather than adding extra `.click()` workarounds.

The important detail in the posted log is that this was running through WebdriverIO's old DevTools/Puppeteer automation path on Firefox. Chrome/Edge worked, while Firefox was sending the first character to browser chrome and then hanging on `elementSendKeys`.

That setup is obsolete for current WebdriverIO. Run Firefox through the normal WebDriver session instead, for example:

```js
capabilities: [{
    browserName: 'firefox'
}]
```

and remove the old DevTools automation configuration.

Current WebdriverIO uses WebDriver/WebDriver BiDi for supported browsers rather than requiring the old Puppeteer-backed Firefox path.

If the same problem can still be reproduced with a current Firefox + Geckodriver + current WDIO release, then it would be worth reducing it to the Google example from the post and reporting it as a Firefox/driver focus issue. But I would not keep the extra `.click()` as the general fix because `setValue()` should target the element without that workaround.

The screenshot URL in the old discussion could no longer be fetched, so I did not rely on it. I did review the posted protocol log and test/step code. There were no replies.

Please mark this answer as accepted if it helped, thank you.
